LDC1.32でCIが動かないためコンパイラバージョンを固定する

LDCのCIが死んでて公開ドキュメント更新できないので、まずLDCだけ動くバージョンに固定してCIを通せるようにします。

切り分けの結果は以下CIで、LDC1.31はOK、LDC1.32はNG、でした。LDCのlatest/master運用はひとまず停止して1.31に固定します。
https://github.com/dlang-jp/Cookbook/actions/runs/4714110512

更新履歴見てみるとdubのビルドキャッシュくらいしか思い当たるものが見つからず原因解消手順が不明です。こちらは必要があればあとで調べることにします。
https://qiita.com/lempiji/items/a7aa1ce5c719dbf578b0#dub-%E5%A4%89%E6%9B%B4%E7%82%B9

ログ消えちゃうので、エラー発生前からのログ抜粋して残しておきます。
Windowsのx86_64、dmd-latestとldc-latestの比較です。

__dmd-latest__

```
2023-04-16T15:43:26.3726552Z Target 'C:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-7TvcoKV9RYEO8Fy90w_C_A\stdx-allocator.lib' doesn't exist, need rebuild.
2023-04-16T15:43:26.3736744Z     Building stdx-allocator 2.77.5: building configuration [library]
2023-04-16T15:43:26.3751491Z [cwd=D:/a/Cookbook/Cookbook] dmd -m64 -lib -ofC:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-7TvcoKV9RYEO8Fy90w_C_A\stdx-allocator.lib -debug -cov -g -w -version=Have_stdx_allocator -IC:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\affix_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\allocator_list.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\bitmapped_block.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\bucketizer.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\fallback_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\free_list.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\free_tree.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\kernighan_ritchie.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\null_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\package.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\quantizer.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\region.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\scoped_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\segregator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\stats_collector.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\common.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\gc_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\internal.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\mallocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\mmap_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\package.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\showcase.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\typed.d -vcolumns
2023-04-16T15:43:26.6861257Z Copying target from C:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-7TvcoKV9RYEO8Fy90w_C_A\stdx-allocator.lib to C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator
2023-04-16T15:43:26.6866790Z Target 'C:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-RE8M0nHNmr_jAdc2XoPHlA\dparse.lib' doesn't exist, need rebuild.
2023-04-16T15:43:26.6882067Z     Building libdparse 0.19.4: building configuration [library]
2023-04-16T15:43:26.6890410Z [cwd=D:/a/Cookbook/Cookbook] dmd -m64 -lib -ofC:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-RE8M0nHNmr_jAdc2XoPHlA\dparse.lib -debug -cov -g -w -version=Have_libdparse -version=Have_stdx_allocator -IC:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src -IC:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\ast.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\astprinter.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\entities.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\formatter.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\lexer.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\parser.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\rollback_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\stack_buffer.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\strings.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\trivia.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\std\experimental\lexer.d -vcolumns
2023-04-16T15:43:28.7340199Z Copying target from C:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-RE8M0nHNmr_jAdc2XoPHlA\dparse.lib to C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse
2023-04-16T15:43:28.7346746Z Target 'C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-mfJinLHp9_xk7THj1k66ww\cookbook-libdparse_usage-test-library.exe' doesn't exist, need rebuild.
2023-04-16T15:43:28.7355883Z     Building cookbook:libdparse_usage ~master: building configuration [cookbook-libdparse_usage-test-library]
2023-04-16T15:43:28.7363763Z [cwd=D:/a/Cookbook/Cookbook] dmd -m64 -c -ofC:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-mfJinLHp9_xk7THj1k66ww\cookbook-libdparse_usage-test-library.obj -debug -cov -g -unittest -w -version=Have_cookbook_libdparse_usage -version=Have_libdparse -version=Have_stdx_allocator -Ithirdparty\libdparse\source -I.github -IC:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\code\cookbook-libdparse_usage-test-library-unittest-cov-z3hgteR8XcCE3IY6unCupQ -IC:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src -IC:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\code\cookbook-libdparse_usage-test-library-unittest-cov-z3hgteR8XcCE3IY6unCupQ\dub_test_root.d thirdparty\libdparse\source\libdparse_usage\example.d .github\ut.d -vcolumns
2023-04-16T15:43:29.7435199Z      Linking cookbook-libdparse_usage-test-library
2023-04-16T15:43:29.7440892Z [cwd=D:/a/Cookbook/Cookbook] dmd -ofC:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-mfJinLHp9_xk7THj1k66ww\cookbook-libdparse_usage-test-library.exe C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-mfJinLHp9_xk7THj1k66ww\cookbook-libdparse_usage-test-library.obj C:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-RE8M0nHNmr_jAdc2XoPHlA\dparse.lib C:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-7TvcoKV9RYEO8Fy90w_C_A\stdx-allocator.lib -m64 -g
```


__ldc-latest__
```
2023-04-16T15:42:58.3344692Z Target 'C:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-F31BB1EC03FFFD5B64AA275D5CBD35138D0014FED955D351FB16C9A108EC72D6\stdx-allocator.lib' doesn't exist, need rebuild.
2023-04-16T15:42:58.3352023Z     Building stdx-allocator 2.77.5: building configuration [library]
2023-04-16T15:42:58.3371439Z ldc2 -march=x86-64 -lib -ofC:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-F31BB1EC03FFFD5B64AA275D5CBD35138D0014FED955D351FB16C9A108EC72D6\stdx-allocator.lib -d-debug -cov -g -w --oq -od=C:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-F31BB1EC03FFFD5B64AA275D5CBD35138D0014FED955D351FB16C9A108EC72D6/obj -d-version=Have_stdx_allocator -IC:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\affix_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\allocator_list.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\bitmapped_block.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\bucketizer.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\fallback_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\free_list.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\free_tree.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\kernighan_ritchie.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\null_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\package.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\quantizer.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\region.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\scoped_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\segregator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\building_blocks\stats_collector.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\common.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\gc_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\internal.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\mallocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\mmap_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\package.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\showcase.d C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source\stdx\allocator\typed.d -vcolumns
2023-04-16T15:42:59.0559061Z Copying target from C:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-F31BB1EC03FFFD5B64AA275D5CBD35138D0014FED955D351FB16C9A108EC72D6\stdx-allocator.lib to C:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator
2023-04-16T15:42:59.0565359Z Target 'C:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-908EA1AE8F47D7BB6C6688B0EB6026B48D65AF5CB3C74CEEA844B0250318EAF3\dparse.lib' doesn't exist, need rebuild.
2023-04-16T15:42:59.0575304Z     Building libdparse 0.19.4: building configuration [library]
2023-04-16T15:42:59.0587250Z ldc2 -march=x86-64 -lib -ofC:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-908EA1AE8F47D7BB6C6688B0EB6026B48D65AF5CB3C74CEEA844B0250318EAF3\dparse.lib -d-debug -cov -g -w --oq -od=C:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-908EA1AE8F47D7BB6C6688B0EB6026B48D65AF5CB3C74CEEA844B0250318EAF3/obj -d-version=Have_libdparse -d-version=Have_stdx_allocator -IC:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src -IC:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\ast.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\astprinter.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\entities.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\formatter.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\lexer.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\parser.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\rollback_allocator.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\stack_buffer.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\strings.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\dparse\trivia.d C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src\std\experimental\lexer.d -vcolumns
2023-04-16T15:43:15.4092911Z Copying target from C:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-908EA1AE8F47D7BB6C6688B0EB6026B48D65AF5CB3C74CEEA844B0250318EAF3\dparse.lib to C:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse
2023-04-16T15:43:15.4103380Z Target 'C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-5CB27F9BA432D4501120A9E9CF112B8CF55C1AC78AB734B415D7AC2806BD72CA\cookbook-libdparse_usage-test-library.exe' doesn't exist, need rebuild.
2023-04-16T15:43:15.4113755Z     Building cookbook:libdparse_usage ~master: building configuration [cookbook-libdparse_usage-test-library]
2023-04-16T15:43:15.4124105Z ldc2 -march=x86-64 -c -ofC:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-5CB27F9BA432D4501120A9E9CF112B8CF55C1AC78AB734B415D7AC2806BD72CA\cookbook-libdparse_usage-test-library.obj -d-debug -cov -g -unittest -w --oq -od=C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-5CB27F9BA432D4501120A9E9CF112B8CF55C1AC78AB734B415D7AC2806BD72CA/obj -d-version=Have_cookbook_libdparse_usage -d-version=Have_libdparse -d-version=Have_stdx_allocator -Ithirdparty\libdparse\source -I.github -IC:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\code\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-912A292B8573C7481E88E48F28958880D5BFE3F209D5A479F59A2D6A28A5DEFC -IC:\Users\runneradmin\AppData\Local\dub\packages\libdparse-0.19.4\libdparse\src -IC:\Users\runneradmin\AppData\Local\dub\packages\stdx-allocator-2.77.5\stdx-allocator\source C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\code\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-912A292B8573C7481E88E48F28958880D5BFE3F209D5A479F59A2D6A28A5DEFC\dub_test_root.d thirdparty\libdparse\source\libdparse_usage\example.d .github\ut.d -vcolumns
2023-04-16T15:43:16.9489909Z      Linking cookbook-libdparse_usage-test-library
2023-04-16T15:43:16.9499536Z ldc2 -ofC:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-5CB27F9BA432D4501120A9E9CF112B8CF55C1AC78AB734B415D7AC2806BD72CA\cookbook-libdparse_usage-test-library.exe C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-5CB27F9BA432D4501120A9E9CF112B8CF55C1AC78AB734B415D7AC2806BD72CA\cookbook-libdparse_usage-test-library.obj C:\Users\runneradmin\AppData\Local\dub\cache\libdparse\0.19.4\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-908EA1AE8F47D7BB6C6688B0EB6026B48D65AF5CB3C74CEEA844B0250318EAF3\dparse.lib C:\Users\runneradmin\AppData\Local\dub\cache\stdx-allocator\2.77.5\build\library-unittest-cov-windows-x86_64-ldc_v1.32.0-F31BB1EC03FFFD5B64AA275D5CBD35138D0014FED955D351FB16C9A108EC72D6\stdx-allocator.lib -march=x86-64 -g
2023-04-16T15:43:17.0288762Z LINK : fatal error LNK1104: cannot open file 'C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-5CB27F9BA432D4501120A9E9CF112B8CF55C1AC78AB734B415D7AC2806BD72CA\cookbook-libdparse_usage-test-library.exe'
2023-04-16T15:43:17.0376376Z Error: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX64\x64\link.exe failed with status: 1104
2023-04-16T15:43:17.0462042Z FAIL C:\Users\runneradmin\AppData\Local\dub\cache\cookbook\~master\+libdparse_usage\build\cookbook-libdparse_usage-test-library-unittest-cov-windows-x86_64-ldc_v1.32.0-5CB27F9BA432D4501120A9E9CF112B8CF55C1AC78AB734B415D7AC2806BD72CA cookbook-libdparse_usage-test-library executable
2023-04-16T15:43:17.0463714Z Error ldc2 failed with exit code 1.
```

--- a/.github/runner.d
+++ b/.github/runner.d
@@ -21,7 +21,8 @@ static:
     immutable subPkgs = [
         PackageInfo("asdf_usage"),
         PackageInfo("libdparse_usage"),
-        PackageInfo("vibe-d_usage", ["windows-x86_omf-", "linux-x86-", "osx-x86-"]),
+        // workaround https://github.com/dlang-jp/Cookbook/issues/198
+        PackageInfo("vibe-d_usage", ["windows-x86_omf-", "linux-x86-", "osx-x86-", "osx-x86_64-"]),
         PackageInfo("windows"),
     ];
 }

--- a/.github/workflows/master.yml
+++ b/.github/workflows/master.yml
@@ -118,33 +118,19 @@ jobs:
           name: coverage-windows
           path: .cov
 
-  test-windows-x86_64-ldc-1_31:
-    name: test-windows-x86_64-ldc-1_31
+  # No. OS      ARCH    COMPILER   UT  DOC COV
+  #  6. Windows x86_64  ldc        o   x   x
+  test-windows-x86_64-ldc-latest:
+    name: test-windows-x86_64-ldc-latest
     runs-on: windows-latest
     steps:
       - uses: actions/checkout@v3
       - name: Install D compiler
         uses: dlang-community/setup-dlang@v1
         with:
-          compiler: ldc-1.31.0
+          compiler: ldc-latest
       - name: Run unit tests
         run: rdmd ./.github/runner.d -a=x86_64 --mode=unit-test
-
-  # No. OS      ARCH    COMPILER   UT  DOC COV
-  #  6. Windows x86_64  ldc        o   x   x
-  # LDC1.32で動かなくなったので一旦停止します
-  # https://github.com/dlang-jp/Cookbook/issues/189
-  # test-windows-x86_64-ldc-latest:
-  #   name: test-windows-x86_64-ldc-latest
-  #   runs-on: windows-latest
-  #   steps:
-  #     - uses: actions/checkout@v3
-  #     - name: Install D compiler
-  #       uses: dlang-community/setup-dlang@v1
-  #       with:
-  #         compiler: ldc-latest
-  #     - name: Run unit tests
-  #       run: rdmd ./.github/runner.d -a=x86_64 --mode=unit-test
 
   # No. OS      ARCH    COMPILER   UT  DOC COV
   #  7. Windows x86_64  dmd-master x   x   x   x
@@ -152,20 +138,18 @@ jobs:
 
   # No. OS      ARCH    COMPILER   UT  DOC COV
   #  8. Windows x86_64  ldc-master o   x   x
-  # LDC1.32で動かなくなったので一旦停止します
-  # https://github.com/dlang-jp/Cookbook/issues/189
-  # test-windows-x86_64-ldc-master:
-  #   name: test-windows-x86_64-ldc-master
-  #   runs-on: windows-latest
-  #   steps:
-  #     - uses: actions/checkout@v3
-  #     - name: Install D compiler
-  #       uses: dlang-community/setup-dlang@v1
-  #       with:
-  #         compiler: ldc-master
-  #         gh_token: ${{ secrets.GITHUB_TOKEN }}
-  #     - name: Run unit tests
-  #       run: rdmd ./.github/runner.d -a=x86_64 --mode=unit-test
+  test-windows-x86_64-ldc-master:
+    name: test-windows-x86_64-ldc-master
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install D compiler
+        uses: dlang-community/setup-dlang@v1
+        with:
+          compiler: ldc-master
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run unit tests
+        run: rdmd ./.github/runner.d -a=x86_64 --mode=unit-test
 
   # No. OS      ARCH    COMPILER   UT  DOC COV
   #  9. Ubuntu  x86     dmd        x   x   x
@@ -321,9 +305,8 @@ jobs:
     needs: [
       test-windows-x86-dmd-latest,
       test-windows-x86_64-dmd-latest,
-      test-windows-x86_64-ldc-1_31,
-      # test-windows-x86_64-ldc-latest,
-      # test-windows-x86_64-ldc-master,
+      test-windows-x86_64-ldc-latest,
+      test-windows-x86_64-ldc-master,
       test-linux-x86-ldc-latest,
       test-linux-x86_64-dmd-latest,
       test-linux-x86_64-ldc-latest,

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -113,34 +113,19 @@ jobs:
         with:
           name: coverage-windows
           path: .cov
-
-  test-windows-x86_64-ldc-1_31:
-    name: test-windows-x86_64-ldc-1_31
+  # No. OS      ARCH    COMPILER   UT  DOC COV
+  #  6. Windows x86_64  ldc        o   x   x
+  test-windows-x86_64-ldc-latest:
+    name: test-windows-x86_64-ldc-latest
     runs-on: windows-latest
     steps:
       - uses: actions/checkout@v3
       - name: Install D compiler
         uses: dlang-community/setup-dlang@v1
         with:
-          compiler: ldc-1.31.0
+          compiler: ldc-latest
       - name: Run unit tests
         run: rdmd ./.github/runner.d -a=x86_64 --mode=unit-test
-
-  # No. OS      ARCH    COMPILER   UT  DOC COV
-  #  6. Windows x86_64  ldc        o   x   x
-  # LDC1.32で動かなくなったので一旦停止します
-  # https://github.com/dlang-jp/Cookbook/issues/189
-  # test-windows-x86_64-ldc-latest:
-  #   name: test-windows-x86_64-ldc-latest
-  #   runs-on: windows-latest
-  #   steps:
-  #     - uses: actions/checkout@v3
-  #     - name: Install D compiler
-  #       uses: dlang-community/setup-dlang@v1
-  #       with:
-  #         compiler: ldc-latest
-  #     - name: Run unit tests
-  #       run: rdmd ./.github/runner.d -a=x86_64 --mode=unit-test
 
   # No. OS      ARCH    COMPILER   UT  DOC COV
   #  7. Windows x86_64  dmd-master x   x   x   x
@@ -148,20 +133,18 @@ jobs:
 
   # No. OS      ARCH    COMPILER   UT  DOC COV
   #  8. Windows x86_64  ldc-master o   x   x
-  # LDC1.32で動かなくなったので一旦停止します
-  # https://github.com/dlang-jp/Cookbook/issues/189
-  # test-windows-x86_64-ldc-master:
-  #   name: test-windows-x86_64-ldc-master
-  #   runs-on: windows-latest
-  #   steps:
-  #     - uses: actions/checkout@v3
-  #     - name: Install D compiler
-  #       uses: dlang-community/setup-dlang@v1
-  #       with:
-  #         compiler: ldc-master
-  #         gh_token: ${{ secrets.GITHUB_TOKEN }}
-  #     - name: Run unit tests
-  #       run: rdmd ./.github/runner.d -a=x86_64 --mode=unit-test
+  test-windows-x86_64-ldc-master:
+    name: test-windows-x86_64-ldc-master
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install D compiler
+        uses: dlang-community/setup-dlang@v1
+        with:
+          compiler: ldc-master
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run unit tests
+        run: rdmd ./.github/runner.d -a=x86_64 --mode=unit-test
 
   # No. OS      ARCH    COMPILER   UT  DOC COV
   #  9. Ubuntu  x86     dmd        x   x   x
@@ -317,9 +300,8 @@ jobs:
     needs: [
       test-windows-x86-dmd-latest,
       test-windows-x86_64-dmd-latest,
-      test-windows-x86_64-ldc-1_31,
-      # test-windows-x86_64-ldc-latest,
-      # test-windows-x86_64-ldc-master,
+      test-windows-x86_64-ldc-latest,
+      test-windows-x86_64-ldc-master,
       test-linux-x86-ldc-latest,
       test-linux-x86_64-dmd-latest,
       test-linux-x86_64-ldc-latest,

--- a/thirdparty/vibe-d/source/vibed_usage/rest.d
+++ b/thirdparty/vibe-d/source/vibed_usage/rest.d
@@ -264,7 +264,7 @@ unittest
             thrown = e;
     });
 
-    auto exitCode = runApplication();
+    auto exitCode = runEventLoop();
     assert(exitCode == 0, "exit code: ".text(exitCode));
     assert(!thrown, thrown.toString());
 

